pkg/ottl/ottlfuncs/func_time.go

```diff
@@ -12,8 +12,9 @@ import (
 )
 
 type TimeArguments[K any] struct {
-	Time   ottl.StringGetter[K]
-	Format string
+	Time     ottl.StringGetter[K]
+	Format   string
+	Location ottl.Optional[string]
```

Comment:
Should this converter support pulling this value from an attribute? If so, this should be `ottl.Optional[ottl.StringGetter]`.

Reply:
i was not thinking about attributes just hardcoded params but attribute seems like a nice option

Reply:
Sorry to go back and forth on this.  I'm looking again and realized we stuck with `string` for `Format` so we could do some validation during startup (enforce non-empty string).  If we kept `Location` a `string`, is there any static validation we could do during collector startup?

Reply:
isn't line 34 what you have in mind for `format` validation? 
in case location is a string we could check if location is a non empty string and if it is present in tzdata database. that's what happening in a function call.

Reply:
> in case location is a string we could check if location is a non empty string and if it is present in tzdata database.

Ya if we could do this on startup we can immediately error if a user entered something like `America/New_Yorks` instead of `America/New_York`.  By adding the `ottl.StringGetter` feature I requested the user won't know they've done something bad in their config until they send data to the collector and then they need to check their logs or, if they are using `error_mode: ignore`, their data to see the transformation didnt happen as they wanted.

I've talked myself back into using `string` instead of `StringGetter` IF we can validate the string location against the database during startup. Sorry for the back and forth.

If we get a solution to https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/29011 we can go to a `StringGetter` and still do startup validation when possible.

Reply:
no worries, just made it a string again and added startup validations as discussed ^^